### DISCUSSION TITLE
Adding C# samples and not just XAML

### DISCUSSION THREAD

**Discussion by CliffAgius** [CommunityToolkit/Maui#359] [New Feature Discussions]
I have been helping with the Docs for the MAUI CT and in these we are adding samples for C# and C# Mark-up and it occurred to me that maybe we should add these samples to the MAUI Sample App.

I'm thinking that in the Sample app you can select XAML/C# or Mark-up and then see all the same pages but developed using that methodology.  The current samples, not just in this app but all samples for Xamarin and MAUI on the web are always XAML UI based, we are creating the sample code for the Docs anyway so this could be a chance to add to the Sample app and it may show Devs how that is built as well.

Just an idea as I know @brminnick is a huge C# UI fan and the more I play with C# for the docs the more I like it.

**Comment by bijington** [comment]
Hey thanks @CliffAgius I like the idea of expanding on how much we are showing for sure! Of course we have the sample over on the markup repository:

https://github.com/CommunityToolkit/Maui.Markup

If we did create further samples it would be nice to think on how we can keep maintenance effort down.

In fact the examples will hopefully move from the conceptual docs pages over to the API docs. Perhaps we could include this as part of the new samples changes

**Comment by CliffAgius** [comment]
Yes I see the sample Markup sample but that is generic where we could have a sample for each of the Converters/Behaviors etc that the user is looking for plus add in the C# samples and it could be a great learning resource.

I agree maintenance is an issue but it already is for the Docs with all 3 samples.

**Reply by bijington** [reply]
This might be where the idea of having the C# section comes back to bite me, originally we were only going to have XAML and C# Markup but I liked showing off how the markup simplifies the C# that can be written and to do that we need the verbose C# section too

**Reply by CliffAgius** [reply]
Yes I realise it's not an easy task but the opportunity for teaching and other learning from the repo is potentially huge as nowhere is this comparison shown hence me thinking in here.

No need for it to be done right away as I know there is still lots to do to catch up and be GA ready but could be a project we add for later when things have calmed down a bit.

**Reply by bijington** [reply]
Yes definitely! I'll aim to bring it up in one of the next stand-ups and see what the rest of the team/community think 👍.

**Comment by CliffAgius** [comment]
Closed as answered.
